src/evaluation/test_validation.py

```diff
@@ -0,0 +1,50 @@
+import pytest
```

Comment:
You add a requirements-dev.txt with `pytest` added.

Reply:
Besides the modifications described above I also converted the repo in a poetry project. The dependencies are now described in the poetry.lock file.